examples/static-auth/README.md

```diff
@@ -0,0 +1,186 @@
+          name: "namespace"
+      resourceAttributes:
+        apiVersion: v1
+        resource: namespace
+        subresource: metrics
+        namespace: "{{ .Value }}"
+      static:
+        - resourceRequest: true
+          resource: namespace
+          subresource: metrics
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: kube-rbac-proxy
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: kube-rbac-proxy
+  template:
+    metadata:
+      labels:
+        app: kube-rbac-proxy
+    spec:
+      securityContext:
+        runAsUser: 65532
+      serviceAccountName: kube-rbac-proxy
+      containers:
+      - name: kube-rbac-proxy
+        image: quay.io/brancz/kube-rbac-proxy:v0.8.0
+        args:
+        - "--secure-listen-address=0.0.0.0:8443"
+        - "--upstream=http://127.0.0.1:8081/"
+        - "--config-file=/etc/kube-rbac-proxy/config-file.yaml"
+        - "--logtostderr=true"
+        - "--v=10"
+        ports:
+        - containerPort: 8443
+          name: https
+        volumeMounts:
+        - name: config
+          mountPath: /etc/kube-rbac-proxy
+        securityContext:
+          allowPrivilegeEscalation: false
+      - name: prometheus-example-app
+        image: quay.io/brancz/prometheus-example-app:v0.1.0
+        args:
+        - "--bind=127.0.0.1:8081"
+      volumes:
+      - name: config
+        configMap:
+          name: kube-rbac-proxy
+```
+
+Once the prometheus-example-app is up and running, we can test it. In order to test it, we deploy a Job, that performs a `curl` against the above deployment. Because it has the correct RBAC roles, the request will succeed.
+
+```bash
+$ kubectl create -f client-rbac.yaml
+```
+
+The content of this manifest is:
+
+[embedmd]:# (./client-rbac.yaml)
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: namespace-metrics
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: namespace-metrics
+subjects:
+- kind: ServiceAccount
+  name: default
+  namespace: default
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: namespace-metrics
+rules:
+- apiGroups: [""]
+  resources:
+  - namespace/metrics
+  verbs: ["get"]
+```
+
+Now simply run
+```
+kubectl run -i -t alpine --image=alpine --restart=Never -- sh -c 'apk add curl; curl -v -s -k -H "Authorization: Bearer `cat /var/run/secrets/kubernetes.io/serviceaccount/token`" https://kube-rbac-proxy.default.svc:8443/metrics?namespace=default'
+```
+
+The full configuration setting for the static authorization feature looks like this:
+```
+  config-file.yaml: |+
+    authorization:
+      static:
+        - user:
+            name: UserName
+            groups:
+              - group1
+              - group2
+          verb: get
+          namespace: default
+          apiGroup: apps
+          resourceRequest: true
+          resource: namespace
+          subresource: metrics
+          path: /metrics
+```
```

Comment:
while this example lists all static authorizer fields, in reality this will never occur. in-flight authorizer Attributes are mutually-excluse either resource request or not, see https://github.com/brancz/kube-rbac-proxy/blob/ea07f61c78044445dc7428e4c1f46360e50fc2e8/pkg/proxy/proxy.go#L151-L181.

Hence, for documentation purposes I suggest to document two examples:

```
  config-file.yaml: |+
    authorization:
      static:
        - user:
            name: UserName
            groups:
              - group1
              - group2
          verb: get
          resourceRequest: false
          path: /metrics
```

and

```
  config-file.yaml: |+
    authorization:
      static:
        - user:
            name: UserName
            groups:
              - group1
              - group2
          verb: get
          namespace: default
          apiGroup: apps
          resourceRequest: false
          resource: namespace
```

Reply:
Yes sounds good. I did consider the mutually exclusivity but forgot to update the docs.

Reply:
i will add this in a follow-up, thanks!